File: development/NET/samples/server/Serial_IO/CS/MyTransaction.cs
using Softing.OPCToolbox;
using Softing.OPCToolbox.Server;

namespace SerialIO
{
	public class MyTransaction : DaTransaction
	{
		#region Public Methods
		//--------------------

		public MyTransaction(
			EnumTransactionType aTransactionType,
			DaRequest[] requestList,
			uint aSessionKey) :
			base(
			aTransactionType,
			requestList,
			aSessionKey)
		{
		}   //	end constructor

		public override int HandleReadRequests()
		{
			//  Handle reads asynchrousouly
			Console.OpcServer.AddRequsts(this.m_requestList);
			return (int)EnumResultCode.S_OK;

		}   //	end HandleReadRequests

		public override int HandleWriteRequests()
		{
			//  Handle writesasynchrousouly

			Console.OpcServer.AddRequsts(this.m_requestList);
			return (int)EnumResultCode.S_OK;

		}   //	end HandleWriteRequests

		//--
		#endregion

	}   //	end class MyTransaction

}   //	end namespace

File: development/NET/samples/server/Serial_IO/VB/MyTransaction.vb
Imports System
Imports Softing.OPCToolbox
Imports Softing.OPCToolbox.Server

Namespace SerialIO

	Public Class MyTransaction
		Inherits DaTransaction

#Region "Public Methods"
		'--------------------

		Public Sub New(ByVal aTransactionType As EnumTransactionType, ByVal requestList As DaRequest(), ByVal aSessionKey As UInt32)
			MyBase.New(aTransactionType, requestList, aSessionKey)
		End Sub

		Public Overloads Overrides Function HandleReadRequests() As Integer
			'  Handle reads asynchrousouly
			Console.OpcServer.AddRequsts(Me.m_requestList)
			Return [Enum].ToObject(GetType(EnumResultCode), EnumResultCode.S_OK)

		End Function

		Public Overloads Overrides Function HandleWriteRequests() As Integer
			'  Handle writesasynchrousouly

			Console.OpcServer.AddRequsts(Me.m_requestList)
			Return [Enum].ToObject(GetType(EnumResultCode), EnumResultCode.S_OK)

		End Function

		'--
#End Region

	End Class

End Namespace

